runelite-client/src/main/java/net/runelite/client/plugins/agility/AgilityPlugin.java

```diff
@@ -421,4 +426,54 @@ private void onTileObject(Tile tile, TileObject oldObject, TileObject newObject)
 			}
 		}
 	}
+
+	@Subscribe
+	public void onMenuEntryAdded(MenuEntryAdded event)
+	{
+		if (!config.showShortcutLevel())
+		{
+			return;
+		}
+
+		if (event.getTarget() == null || event.getTarget() == "")
+		{
+			return;
+		}
+
+		//Guarding against non-first option because agility shortcuts are always that type of event.
+		if (event.getType() != MenuAction.GAME_OBJECT_FIRST_OPTION.getId())
+		{
+			return;
+		}
+
+		MenuEntry[] menuEntries = client.getMenuEntries();
+
+		for (Obstacle nearbyObstacle : getObstacles().values())
```

Comment:
We can greatly reduce nesting & complexity here with streams and functional programming.

Untested but this logic could look something like this:
```
final int entryId = event.getIdentifier();
MenuEntry[] menuEntries = client.getMenuEntries();
for (Obstacle obstacle : getObstacles().values())
{
	AgilityShortcut shortcut = obstacle.getShortcut();
	if (shortcut.getObstaclesIds().stream().anyMatch(obstacleId -> entryId == obstacleId))
	{
		int level = shortcut.getLevel();
		String color = level <= getAgilityLevel() ? Color.GREEN : Color.RED;
		String requirementText = " (level-" + level + ")";
		MenuEntry entry = menuEntries[menuEntries.length - 1];
		entry.setTarget(event.getTarget() + ColorUtil.prependColorTag(requirementText, color));
	}
}
client.setMenuEntries(menuEntries);
```